pkg/apis/pravega/v1alpha1/status.go

```diff
@@ -0,0 +1,136 @@
+	PravegaClusterConditionError                               = "Error"
+)
+
+// PravegaClusterStatus defines the observed state of PravegaCluster
+type PravegaClusterStatus struct {
+	// Conditions list all the applied conditions
+	Conditions []PravegaClusterCondition `json:"conditions,omitempty"`
+
+	// CurrentVersion is the current cluster version
+	CurrentVersion string `json:"currentVersion"`
+
+	// TargetVersion is the version the cluster upgrading to.
+	// If the cluster is not upgrading, TargetVersion is empty.
+	TargetVersion string `json:"targetVersion"`
+}
+
+// PravegaClusterCondition shows the current condition of a Pravega cluster.
+// Comply with k8s API conventions
+type PravegaClusterCondition struct {
+	// Type of Pravega cluster condition.
+	Type PravegaClusterConditionType `json:"type"`
+
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+
+	// The reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+
+	// A human readable message indicating details about the transition.
+	Message string `json:"message,omitempty"`
+
+	// The last time this condition was updated.
+	LastUpdateTime string `json:"lastUpdateTime,omitempty"`
+
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime string `json:"lastTransitionTime,omitempty"`
+}
+
+func (ps *PravegaClusterStatus) ContainsCondition(condType PravegaClusterConditionType) bool {
+	if _, conditon := ps.getClusterCondition(condType); conditon != nil {
+		return true
+	}
+	return false
+}
+
+func (ps *PravegaClusterStatus) SetReadyCondition() {
+	c := newClusterCondition(PravegaClusterConditionReady, corev1.ConditionTrue, "Cluster available", "")
+	ps.setClusterCondition(*c)
+}
+
+func (ps *PravegaClusterStatus) SetScalingCondition() {
+	c := newClusterCondition(PravegaClusterConditionScaling, corev1.ConditionTrue, "Cluster scaling", "")
+	ps.setClusterCondition(*c)
+}
+
+func (ps *PravegaClusterStatus) SetErrorCondition(message string) {
+	c := newClusterCondition(PravegaClusterConditionScaling, corev1.ConditionTrue, "Cluster error", message)
+	ps.setClusterCondition(*c)
+}
+
+func (ps *PravegaClusterStatus) ClearReadyCondition() {
+	ps.deleteClusterCondition(PravegaClusterConditionReady)
+}
+
+func (ps *PravegaClusterStatus) ClearScalingCondition() {
+	ps.deleteClusterCondition(PravegaClusterConditionScaling)
+}
+
+func (ps *PravegaClusterStatus) ClearErrorCondition() {
+	ps.deleteClusterCondition(PravegaClusterConditionError)
+}
+
+func newClusterCondition(condType PravegaClusterConditionType, status corev1.ConditionStatus, reason, message string) *PravegaClusterCondition {
+	now := time.Now().Format(time.RFC3339)
+	return &PravegaClusterCondition{
+		Type:               condType,
+		Status:             status,
+		Reason:             reason,
+		Message:            message,
+		LastUpdateTime:     now,
+		LastTransitionTime: now,
```

Comment:
`LastTransitionTime` should only be set to `now` when the condition status changes from the previous value.

Reply:
Got it